### PR TITLE
Use CacheControlScope.Public per default

### LINKED ISSUE #7055: HotChocolate.Caching with queries using DefaultMaxAge will be scoped to private
### Product

Hot Chocolate

### Version

13.8.1

### Link to minimal reproduction

https://github.com/kiangkuang/HotChocolateDemo

### Steps to reproduce

1. Clone this project that has been minimally configured like in https://www.youtube.com/watch?v=sk4XD3Zf64k
2. Run the project and execute the `bookWithDefaultCacheControl` query, which has no `CacheControlAttribute` and will use the value from `DefaultMaxAge` defined in https://github.com/kiangkuang/HotChocolateDemo/blob/1e3239dd7d37ef78ab9cc5cc5fabecd7912cf1be/Program.cs#L9
    ```
    query {
        bookWithDefaultCacheControl {
            title
        }
    }
    ````

### What is expected?

Response with header 
1. `Cache-Control: public, max-age=20000` so the behaviour is consistent with the `CacheControlAttribute.Scope` default value, or
2. `Cache-Control: max-age=20000` since no default scope was defined, or
3. A way to configure a default scope like `DefaultMaxAge`

### What is actually happening?

Response with header `Cache-Control: private, max-age=20000`, with no way to change or remove the `private` scope for queries without `CacheControlAttribute` and using the `DefaultMaxAge`.

This makes the option of `DefaultMaxAge` unusable for projects where we want the default scope to be public.

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tobias-tengler**
Fixes #7055

### CODE REVIEW

File: src/HotChocolate/Caching/src/Caching/Types/CacheControlScope.cs [hunk 1]
```diff
@@ -5,17 +5,17 @@ namespace HotChocolate.Caching;
...
     /// therefore be stored in a shared cache.
     /// </summary>
     Public,
+
+    /// <summary>
+    /// Private query results contain data
+    /// that is scoped to a user and must therefore
+    /// be only cached for and accessible to that user.
+    /// </summary>
+    Private,
```
Comment (tobias-tengler) [inline src/HotChocolate/Caching/src/Caching/Types/CacheControlScope.cs:20]: The problem is that parsing a directive currently always initializes the properties of the runtime directive type, EVEN if they are nullable.
This is just a temporary fix that also causes other caching functionality to fail. I'm still investigating the underlying issue.

**Comment by kiangkuang** [comment]
Would it be more appropriate to allow the default scope be customizable similar to `DefaultMaxAge`, or should that be deferred to a future enhancement?

**Comment by tobias-tengler** [comment]
Yes, this can also be done, but it wouldn't address the underlying issue. I'm going to add it later.

File: src/HotChocolate/Core/src/Types/Types/Argument.cs [hunk 2]
```diff
@@ -103,6 +103,12 @@ public Argument(ArgumentDefinition definition, int index)
 
         Type = context.GetType<IInputType>(definition.Type!).EnsureInputType();
         _runtimeType = definition.RuntimeType ?? definition.Parameter?.ParameterType!;
+
+        if (definition is DirectiveArgumentDefinition { Property.PropertyType: {} propertyType })
+        {
+            _runtimeType = propertyType;
+        }
```
Comment (tobias-tengler) [inline src/HotChocolate/Core/src/Types/Types/Argument.cs:110]: @michaelstaib At first I wasn't sure whether this was the best solution, but after looking deeper into it, I now believe that this is the only (feasible) solution atm.

Comment (michaelstaib) [reply src/HotChocolate/Core/src/Types/Types/Argument.cs:110]: I do not think so :D

Comment (michaelstaib) [reply src/HotChocolate/Core/src/Types/Types/Argument.cs]: OK ... reworked this one.

**Review by michaelstaib — APPROVED** [review]

**Pull request merged**